### ISSUE TITLE
New Project: mobanise tool

### ISSUE THREAD

**Issue by jayvdb** [coala/projects#574]
The process of converting a project to moban is similar to the coala-quickstart problem solved in the [GSoC 2017 quickstart project](https://github.com/coala/projects/blob/master/_projects/enhance_coala_quickstart.md)

A lot of information needs to be extracted from various existing files, converted to yaml, and then moban run to generate those same files, and various commands run to identify what functional changes have been made by using the moban templates, and a human checking those changes to see if they were good.

Before this project is started, the info extraction components from quickstart need to be moved to a new library.  Ideally that is done by the authors, with the history of the code retained in the new repository.

Then this project would involve improving that library, and creating a new command `mobanise` that uses the library to generate at least the two main files of moban, a `.moban.yaml` & `<project name>.yaml`.

https://github.com/coala/coala-quickstart/tree/master/coala_quickstart/info_extraction
https://github.com/coala/coala-quickstart/tree/master/coala_quickstart/info_extractors

The new tool should work on a variety of existing Python projects, especially hand-rolled setup.py that are not too complicated, `pbr` based projects, and including those using other systems such as `versioneer`.

cc @chfw

**Comment by satwikkansal** [comment]
Yeah, we can move them to coalib or maybe a separate repository called `info-parsers`. Relevant: https://github.com/coala/coala-quickstart/issues/117

**Comment by Naveenaidu** [comment]
@jayvdb The link to  `GSoC 2017 quickstart project` is dead. Can you please update it.

**Comment by chfw** [comment]
I can help if people in SoC 2019 will get stuck with mobanising any projects.

**Comment by chfw** [comment]
Instead of mobanise tool, could I propose an alternative?

it could be an ‘yehua.yaml’ + a list of static files and template files + individual template repos. Take yehua’s ability for standard python package as an example, it has pupil-mob and-pkg as template repo, and embeds a yehua file and a few stater files. Pyexcel also got similar setup for its plugins. It is similar to Cookiecutter’s approach.

https://github.com/moremoban/yehua
